### ISSUE TITLE
Large Pandas dataframes are resulting in an Empty Query Error

### ISSUE THREAD

**Issue by maharshi365** [ClickHouse/clickhouse-connect#258]
### Describe the bug
When trying to insert large pandas dataframes, the client throws an error saying Empty Query. I am able to workaround this by manually breaking the dataframe and then looping over to insert.

### Steps to reproduce
1. Create table with statement below
2. Try to insert large dataframe

### Expected behaviour
The entire dataframe should be inserted in one go without errors.

### Code example
```python

ch_client.insert_df(
            df=df,
            table_name="transactions",
            columns=list(df.columns),
)
```

### clickhouse-connect and/or ClickHouse server logs

```
Traceback (most recent call last):
  File "C:\Users\me\OneDrive\Desktop\project\src\indexer\src\indexers\evm_indexer.py", line 107, in _process_chunk
    rows = processor.write_to_ch(processed_df, self.ch, chunk)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\me\OneDrive\Desktop\project\src\indexer\src\processors\evm\ethereum\transactions.py", line 130, in write_to_ch
    inserted = ch_client.insert_df(
               ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\me\OneDrive\Desktop\project\src\indexer\src\db\ch_manager.py", line 29, in insert_df
    res = self.client.insert_df(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\me\OneDrive\Desktop\project\env\Lib\site-packages\clickhouse_connect\driver\client.py", line 585, in insert_df
    return self.insert(table,
           ^^^^^^^^^^^^^^^^^^
  File "C:\Users\me\OneDrive\Desktop\project\env\Lib\site-packages\clickhouse_connect\driver\client.py", line 554, in insert
    return self.data_insert(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\me\OneDrive\Desktop\project\env\Lib\site-packages\clickhouse_connect\driver\httpclient.py", line 249, in data_insert
    response = self._raw_request(block_gen, params, headers, error_handler=error_handler, server_wait=False)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\me\OneDrive\Desktop\project\env\Lib\site-packages\clickhouse_connect\driver\httpclient.py", line 430, in _raw_request
    error_handler(response)
  File "C:\Users\me\OneDrive\Desktop\project\env\Lib\site-packages\clickhouse_connect\driver\httpclient.py", line 236, in error_handler
    self._error_handler(resp)
  File "C:\Users\me\OneDrive\Desktop\project\env\Lib\site-packages\clickhouse_connect\driver\httpclient.py", line 356, in _error_handler
    raise OperationalError(err_str) if retried else DatabaseError(err_str) from None
clickhouse_connect.driver.exceptions.DatabaseError: :HTTPDriver for https://f8e7lc7q7k.us-east-2.aws.clickhouse.cloud:8443 returned response code 400)
 Code: 62. DB::Exception: Empty query. (SYNTAX_ERROR) (version 23.8.1.41588 (official build))
```

#### Environment
* clickhouse-connect version: 0.6.14
* pandas version: 2.0.3
* Python version: 3.11.4
* Operating system: Windows 11

#### ClickHouse server
* ClickHouse Server version: Clickhouse Cloud
* ClickHouse Server non-default settings, if any:
* `CREATE TABLE` statements for tables involved:
```
-- Receipts Table
CREATE TABLE IF NOT EXISTS ethereum_transactions (
  blockNumber UInt64,
  blockHash String,
  transactionHash String DEFAULT '',
  transactionIndex UInt64 DEFAULT -1,
  timestamp DateTime,
  cumulativeGasUsed UInt32 DEFAULT 0,
  effectiveGasPrice UInt64 DEFAULT 0,
  gas UInt32 DEFAULT 0,
  gasUsed UInt32 DEFAULT 0,
  gasPrice UInt64 DEFAULT 0,
  maxPriorityFeePerGas UInt64 DEFAULT 0,
  maxFeePerGas UInt64 DEFAULT 0,
  from String DEFAULT '',
  to String DEFAULT '',
  input String DEFAULT '0x',
  value Decimal(38, 0),
  contractAddress String DEFAULT '',
  type Nullable(UInt8),
  status Nullable(UInt8)
) ENGINE = MergeTree()
ORDER BY (to, timestamp);
-- transactions from Projection
ALTER TABLE ethereum_transactions
ADD PROJECTION ethereum_transactions_from (
    select *
    ORDER BY (
        from,
          timestamp
      )
  );
-- transactions Table Materialize Projection
Alter table ethereum_transactions materialize projection ethereum_transactions_from;
```
* Sample data for these tables, use [clickhouse-obfuscator](https://github.com/ClickHouse/ClickHouse/blob/master/programs/obfuscator/Obfuscator.cpp#L42-L80) if necessary

Sample Data: [Google Drive Link](https://drive.google.com/file/d/14XKiL9K8wG0BV8RnJA7wGb5HaryjBWGl/view?usp=sharing)

**Comment by genzgd** [comment]
This should be fixed in the 0.6.18 release.  If you could retest in your environment that would be much appreciated.  I'm not exactly sure why there's a size limit on the inserted chunks, but I've reduced the chunk block size which fixed the problem uploading this parquet file/DataFrame to ClickHouse cloud.

**Comment by maharshi365** [comment]
can confirm this works. Thanks @genzgd!
